tests/test_routing_paths.py

```
@@ -0,0 +1,150 @@
+        confidence = asyncio.run(router._assess_confidence(low_conf_response, request))
+        assert confidence < 0.5
+    
+    @pytest.mark.asyncio
+    async def test_support_threshold_routing(self, router):
+        """Test routing based on RAG support threshold."""
+        
+        # Mock low RAG support
+        with patch.object(router, '_get_rag_support_score', return_value=0.3):
+            # Even simple question should route to thinking if low RAG support
+            decision = await router._determine_routing(GenRequest(prompt="Simple question"))
+            assert decision.provider == "deepseek_thinking"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)